code/game/gamemodes/wizard/godhand.dm

```diff
@@ -32,6 +32,9 @@
 	attached_spell.attached_hand = null
 	qdel(src)
 
+/obj/item/weapon/melee/touch_attack/dropped()
+	qdel(src)
```

Comment:
Actually, it would probably be cleaner to just move the canceling code to a new proc on the spell and call that from here.